### PR TITLE
feat: add bulk actions to HCMS

### PR DESCRIPTION
**Pull request by leopuleo**
## Changes
With this PR, we allow bulk actions to headless CMS entries. The default bulk actions are:

- Publish
- Unpublish
- Move
- Delete

Third-party developers can add bulk actions: a new article describing the process will be added to the developer docs.

## How Has This Been Tested?
Jest + manually

## Documentation
Inline + Developer Doc article (TBD)

### CODE REVIEW

File: packages/app-admin/src/components/Buttons/Buttons.tsx [hunk 1]
```diff
@@ -34,7 +36,7 @@ export const Buttons: React.VFC<ButtonsProps> = props => {
     return (
         <>
             {props.actions.map(action => (
-                <ButtonContainer key={action.name}>
+                <ButtonContainer key={action.name} className={"button-container"}>
```
Comment (Pavel910) [inline packages/app-admin/src/components/Buttons/Buttons.tsx]: Adding CSS classes randomly is a bad idea. It's better to either add props to the component,  or simply create a component variant, like `<NoMarginButtonContainer>`,  and you can even keep it local to this view. 

```tsx
const NoMarginButtonContainer = styled(ButtonContainer)`
    margin: 0;
`;
```

File: packages/app-admin/src/components/BulkActions/Worker.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     * @param callback - The callback function to process each item.
+     * @param {T[]} chunk - The chunk of items to be processed.
+     * @returns {Promise<void>}
+     */
+    private async processChunk(
+        callback: ({ item, allItems, report }: CallbackParams<T>) => void | Promise<void>,
+        chunk: T[]
+    ): Promise<void> {
+        for (const item of chunk) {
+            await callback({ item, allItems: this._items, report: this._report });
```
Comment (Pavel910) [inline packages/app-admin/src/components/BulkActions/Worker.ts:135]: Do you think it's work adding `pRetry` here, or do you want the developer to handle failed executions/api calls in their callback?

File: packages/app-headless-cms/src/admin/components/ContentEntries/BulkActions/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    padding: 0 16px;
+`;
+
+export const ButtonsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    .button-container {
+        margin: 0;
+    }
```
Comment (Pavel910) [inline packages/app-headless-cms/src/admin/components/ContentEntries/BulkActions/styles.tsx]: This is connected to my other comment on randomly adding classes. Nobody knows this exists, it's not easily discoverable, it's not in TS autocomplete, so let's avoid these hidden magic classes.

**Review by Pavel910 — CHANGES_REQUESTED** [review]
Just a few comments here, and we are good. In general, it's great that you developed a system first (the configuration), then followed the same approach for every action.

File: packages/app-admin/src/components/Buttons/Buttons.tsx (hunk 1 above)
Comment (leopuleo) [reply packages/app-admin/src/components/Buttons/Buttons.tsx]: Here is the problem: We are using the same component to render [Entry Editor Actions](https://www.webiny.com/docs/headless-cms/extending/customize-entry-editor-actions) and now the Entry List Bulk actions. 

I don't like to use the CSS class but I need to target the single action from different parts of the UI.

I changed it: now I'm exporting the styled component from the `@webiny/app-admin` package so I can target it inside the specific component.

File: packages/app-admin/src/components/BulkActions/Worker.ts (hunk 2 above)
Comment (leopuleo) [reply packages/app-admin/src/components/BulkActions/Worker.ts:135]: I love having a retry strategy but would not bake it inside the Worker, keeping it simple and straightforward. 

I noticed that we are already exposing a helper function that wraps [pRetry](https://github.com/webiny/webiny-js/blob/next/packages/utils/src/executeWithRetry.ts)...developer could use it outside the Worker logic.

**Review by Pavel910 — APPROVED** [review]

**Pull request merged**